### PR TITLE
[SELS-TASK][FE/BE] Sign Up Page and Functions 

### PR DESCRIPTION
**Pull request by michaelsilverio**
### [Asana Link]
[link here](https://app.asana.com/0/1201713838250812/1201726038594791/f)

### [Commands]
```
# added new fields to users table
php artisan migrate:fresh --seed

php artisan serve
npm run watch
```

### [Pre-conditions]
After running `npm run watch` and `php artisan serve`. Go to the url being generated on artisan serve

### [Expected Output]
Details

- On field inputs or leaving fields blank, should display an error below the field
- The password field visibility toggle should work
- The confirm password visibility toggle should work
- Checks if invalid email then display error message
- Password will display an error if below 8 characters
- Submit is disabled when fields have errors
- On submit, if email already exists, display "The email has already been taken." error message
- On submit, if successful, displays "Successfully Registered" then navigate to sign in page after 2.5 seconds

### [Notes]
- The email and password field functionality in Sign In page is the same in the Sign Up page. This is to reuse the code.
- There is a function `freshState()` in props that re-initializes the state values. This is helpful so that every time you go back and forth in Sign In or Sign Up page, the state values are in default. 
### [Screenshots]
image
image
image

### CODE REVIEW

File: resources/js/actions/userAuthActions.js [hunk 1]
```diff
@@ -1,244 +1,270 @@
...
+      confirmPass !== ''
+    ) {
+      await userApi
+        .post('/sign-up', users)
+        .then((response) => {
+          data = {
+            requestError: response.data.error,
+            requestErrorMessage: response.data.message
+          };
+          console.log(data);
```
Comment (AubreyV) [inline resources/js/actions/userAuthActions.js]: remove console.log here

File: resources/js/components/SignIn.js [hunk 2]
```diff
@@ -1,292 +1,250 @@
...
-        /* 
-            When navigated to this component, the password should be cleared
-            but when you put password in this component and try to navigate to
-            Sign Up, the password field has values. This checks if password has
-            values on first render then clear the values.
-        */
-    }, []);
+  // Check requestError, if false then navigate to dashboard page
+  useEffect(() => {
+    if (props.requestError === false) {
```
Comment (AubreyV) [inline resources/js/components/SignIn.js:38]: no need to explicitly add `=== false` if `props.requestError`'s value is boolean, you can just do `if (props.requestError)`

File: resources/js/components/SignIn.js [hunk 3]
```diff
@@ -1,292 +1,250 @@
...
-            const timer = setTimeout(() => {
-                navigate("/dashboard");
-            }, 2500);
-            return () => clearTimeout(timer);
-        }
-    }, [props.requestError]);
+  // This will run, if there is changes in the state
+  useEffect(() => {
+    // Enable the submit button if no errors in fields
+    if (props.isValidEmail === true && props.isValidPassword === true) {
```
Comment (AubreyV) [inline resources/js/components/SignIn.js]: here as well, no need to explicitly check if `true` or `false` if variable's value is boolean.

```suggestion
    if (props.isValidEmail && props.isValidPassword) {
```

File: resources/js/components/SignIn.js [hunk 4]
```diff
@@ -1,292 +1,250 @@
...
-                        onSubmit={handleSubmit}
-                        sx={{ marginTop: "10px" }}
+  // Toggle Show Password
+  const handleShowPassword = () => {
+    props.showPassword(props.isShownPass);
+  };
+
+  return (
+    <Grid container component="main" sx={{ height: '100vh' }}>
+      {props.requestError === false ? (
```
Comment (AubreyV) [inline resources/js/components/SignIn.js:76]:
```suggestion
      {props.requestError ? (
```

File: resources/js/components/SignIn.js [hunk 5]
```diff
@@ -1,292 +1,250 @@
...
+              autoComplete="off"
+            />
+            <TextField
+              onKeyUp={(e) => {
+                props.validatePassword('signIn', e.target.value);
+              }}
+              helperText={
+                props.passwordError === '' ? ' ' : props.passwordError
+              }
+              error={props.passwordError === '' ? false : true}
```
Comment (AubreyV) [inline resources/js/components/SignIn.js:169]:
```suggestion
              error={props.passwordError === ''}
```

File: resources/js/actions/userAuthActions.js (hunk 1 above)
Comment (michaelsilverio) [reply resources/js/actions/userAuthActions.js]: Remove the console.log(data)

File: resources/js/components/SignIn.js (hunk 2 above)
Comment (michaelsilverio) [reply resources/js/components/SignIn.js:38]: When removing the `=== false`  makes the redirection after successful request don't work.

File: resources/js/components/SignIn.js (hunk 3 above)
Comment (michaelsilverio) [reply resources/js/components/SignIn.js]: Removed the `=== true` and it is still functional. This is resolved

File: resources/js/components/SignIn.js (hunk 4 above)
Comment (michaelsilverio) [reply resources/js/components/SignIn.js:76]: removing the `=== false` will make the Sign In dialog not show.

File: resources/js/components/SignIn.js (hunk 5 above)
Comment (michaelsilverio) [reply resources/js/components/SignIn.js:169]: Removing the `=== '' ? false : true` the error will not highlight the text field as error. `error={}` should be a true or false.

File: resources/js/components/SignIn.js (hunk 2 above)
Comment (AubreyV) [reply resources/js/components/SignIn.js:38]: @michaelsilverio okay disregard the comment na lang.

File: resources/js/components/SignIn.js (hunk 4 above)
Comment (AubreyV) [reply resources/js/components/SignIn.js:76]: @michaelsilverio oki disregard the comment na lang

File: resources/js/components/SignIn.js (hunk 5 above)
Comment (AubreyV) [reply resources/js/components/SignIn.js:169]: @michaelsilverio oki u can disregard this as well

**Review by AubreyV — APPROVED** [review]
## Approved
1. Sign Up working as expected
2. Can register a user with validations
3. Can login user with correct credentials
4. Shows error if password is wrong

**Pull request merged**